lib/r10k/action/runner.rb

```diff
@@ -42,10 +42,23 @@ def setup_logging
       def setup_settings
         config_settings = settings_from_config(@opts[:config])
 
-        overrides = {:cachedir => @opts[:cachedir]}
-        overrides.delete_if { |_, val| val.nil? }
+        overrides = {
+          cachedir: @opts[:cachedir],
+          deploy: {
+            puppet_path: @opts[:'puppet-path'],
+            generate_types: @opts[:'generate-types']
+          }
+        }
 
         with_overrides = config_settings.merge(overrides) do |key, oldval, newval|
+          break oldval if newval.nil?
```

Comment:
I'm currently seeing all non-trivial r10k actions failing with `NoMethodError: undefined method `keys' for "/var/cache/r10k":String`, I think this line is the culprit.

AFAICT, `break` will exit the merge loop here, causing `with_overrides` to be whatever the value of `oldval` is. eg. the "/var/cache/r10k" instead of the config_settings hash.

Reply:
I've been using this code in my environments for months, but admittedly in the 2.6.x branch. I think you're right that this should have been a `next` instead of `break`. Does r10k work correctly for you if you make that change?

Reply:
@nabertrand tests were passing before I merged it. Think you could update tests to catch this problem too?

Reply:
I confirmed that `break` is causing the problem @justinstoller is seeing and will add some tests to catch this in the future.

Reply:
@binford2k #917 should fix this. Interestingly, this PR would have failed the spec tests if they had run more recently. It would be nice if the Travis CI/azure checks automatically ran again if the target branch changed.